crates/matrix-sdk/src/sliding_sync/list/builder.rs

```diff
@@ -155,15 +148,15 @@ impl SlidingSyncListBuilder {
             SlidingSyncMode::Selective => SlidingSyncListRequestGenerator::new_selective(),
         };
 
-        Ok(SlidingSyncList {
+        SlidingSyncList {
             inner: Arc::new(SlidingSyncListInner {
                 // From the builder
                 sync_mode: self.sync_mode,
                 sort: self.sort,
                 required_state: self.required_state,
                 filters: self.filters,
                 timeline_limit: StdRwLock::new(Observable::new(self.timeline_limit)),
-                name: self.name.ok_or(Error::BuildMissingField("name"))?,
```

Comment:
Can we remove this `BuildMissingField` variant from the enum?

Reply:
No unfortunately, it's still used in the `SlidingSyncBuilder::build()` method when the client is missing. I could probably go ahead and try to introduce the same change there.

Reply:
I could do just that, `SlidingSyncBuilder::build` is still fallible because it's the point where it reads from the cache and that's fallible.